src/mlpack/tests/rectangle_tree_test.cpp

```diff
@@ -1350,3 +1350,77 @@ TEST_CASE("RectangleTreeMoveDatasetTest", "[RectangleTreeTraitsTest]")
   REQUIRE(tree.Dataset().n_rows == 3);
   REQUIRE(tree.Dataset().n_cols == 1000);
 }
+
+// Test that points can be added to the tree.
+TEST_CASE("RectangleTreeInsertVsBatchTest", "[RectangleTreeTest]")
+{
+  // Make sure that manually inserting points gives the same tree as if we build
+  // the tree all at once.
+  arma::mat dataset = arma::randu<arma::mat>(3, 1000);
+  using TreeType = RTree<EuclideanDistance, EmptyStatistic, arma::mat>;
+
```

Comment:
Great to have this one, I was looking for it ?

Reply:
Yeah, I was surprised it was not written earlier, actually.